src/App.tsx

```diff
@@ -1,18 +1,62 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const total = 42;
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const numberOfPages = Math.ceil(total / perPage);
+
+  const allPages = getNumbers(1, numberOfPages);
+
+  let items: string[] = [];
+
+  if (allPages.length !== currentPage) {
+    switch (currentPage) {
+      case 1:
+        items = getNumbers(1, perPage)
+          .map(n => `Item ${n}`);
+
+        break;
+
+      case 2:
+      case 3:
+      case 4:
+      case 5:
+      case 6:
+      case 7:
+      case 8:
+      case 9:
+      case 10:
+      case 11:
+      case 12:
+      case 13:
+      case 14:
+        items = getNumbers(perPage * (currentPage - 1) + 1, perPage
+        * currentPage)
+          .map(n => `Item ${n}`);
+
+        break;
+
+      default: break;
+    }
+  }
```

Comment:
strange logic for items on page change logic, consider setting a state and storing/setting necessary values there and just use state where needed